evaluator/functions.go

```diff
@@ -1319,3 +1346,57 @@ func sleepFn(tok token.Token, args ...object.Object) object.Object {
 
 	return NULL
 }
+
+// source("fileName")
+// aka src(), include(), import(), require()
+var sourceMaxRecursionDepth = 5
```

Comment:
what is this? We cannot require a file that requires a file that requires a file etc etc more than 5 levels down? If so, I'd be against this -- once you build your own userland library you might have quite a few levels of nested dependencies...

Reply:
The default source file recursion limit can be set really high (say 100) and/or we can create an env var that allows the user to declare the intended depth. Either way we need to protect against unintended circular inclusions that will panic the ABS interpreter.

For example, `ABS_SOURCE_DEPTH=15` could be set either in the OS env or the ABS Init File or just prior to using a `source()` statement that is known to have a given depth. Thus,
```
ABS_SOURCE_DEPTH = 15
source("path/to/lib")
```
If this these lines are in the ABS Init File, then the `ABS_SOURCE_DEPTH` will apply to future `source()` statements until it is overridden.